spec/api/v1/external_users/claims/integration/litigator_claim_creation_spec.rb

```diff
@@ -226,229 +441,24 @@ def results(claim, expected)
-      let(:representation_order_date) { Date.new(2018, 03, 31).as_json }
-
-      specify 'Case management system creates a valid scheme 9 fixed fee claim' do
-        post ClaimApiEndpoints.for(:final).create, claim_params, format: :json
-        expect(last_response.status).to eq 201
-
-        claim = Claim::BaseClaim.find_by(uuid: last_response_uuid)
-
-        post endpoint(:defendants), defendant_params.merge(claim_id: claim.uuid), format: :json
-        expect(last_response.status).to eq 201
-
-        defendant = Defendant.find_by(uuid: last_response_uuid)
-
-        post endpoint(:representation_orders), representation_order_params.merge(defendant_id: defendant.uuid), format: :json
-        expect(last_response.status).to eq 201
+    it_behaves_like 'LGFS API claim creation'
+  end
 
-        post endpoint(:fees), fixed_fee_params_with_amount.merge(claim_id: claim.uuid, fee_type_id: fixed_fee_type.id), format: :json
-        expect(last_response.status).to eq 201
+  context 'scheme 10' do
+    let(:representation_order_date) { Date.new(2022, 9, 30).as_json }
+    let(:fee_scheme) { ['LGFS', 10] }
+    let(:date_param) { '2022-10-01' }
 
-        expect(claim.fixed_fees.size).to eq 1
-        expect(claim.fixed_fees.first.quantity).to eq 1
-        expect(claim.fixed_fees.first.rate).to eq 349.47
-        expect(claim).to be_valid_api_lgfs_claim(fee_scheme: ['LGFS', 9], offence: nil, total: 349.47, vat_amount: 69.89)
+    around do |example|
+      travel_to(Settings.lgfs_scheme_10_clair_release_date + 1.month) do
+        example.run
       end
     end
 
-    context 'hardship claim' do
-      let(:case_type) { nil }
-      let(:case_stage) { create(:case_stage, :pre_ptph_or_ptph_adjourned) }
-      let(:representation_order_date) { Date.new(2020, 03, 31).as_json }
-
-      specify 'Case management system creates a valid hardship claim' do
-        post ClaimApiEndpoints.for('litigators/hardship').create, claim_params.merge(case_stage_unique_code: case_stage.unique_code, offence_id: offence.id), format: :json
-        expect(last_response.status).to eq 201
-
-        claim = Claim::BaseClaim.find_by(uuid: last_response_uuid)
-
-        post endpoint(:defendants), defendant_params.merge(claim_id: claim.uuid), format: :json
-        expect(last_response.status).to eq 201
-
-        defendant = Defendant.find_by(uuid: last_response_uuid)
-
-        post endpoint(:representation_orders), representation_order_params.merge(defendant_id: defendant.uuid), format: :json
-        expect(last_response.status).to eq 201
-
-        post endpoint(:fees), graduated_fee_params.merge(claim_id: claim.uuid, fee_type_id: graduated_fee_type.id, date: representation_order_date), format: :json
-        expect(last_response.status).to eq 201
-
-        expect(claim).to be_valid_api_lgfs_claim(fee_scheme: ['LGFS', 9], offence:, total: 5142.87, vat_amount: 1028.57)
-      end
-    end
+    it_behaves_like 'LGFS API claim creation'
```

Comment:
A couple of things you may consider (up to you);

1) I think, if you want, you can do this:

```ruby
    it_behaves_like 'LGFS API claim creation' do
      let(:representation_order_date) { Date.new(2022, 9, 30).as_json }
      let(:fee_scheme) { ['LGFS', 10] }
      let(:date_param) { '2022-10-01' }

      around do |example|
        travel_to(Settings.lgfs_scheme_10_clair_release_date + 1.month) do
          example.run
        end
      end
```

(and likewise for the scheme 9 version, above)


2) Also, instead of `around` I prefer something like this if possible;

```ruby
    before { travel_to(Settings.lgfs_scheme_10_clair_release_date + 1.month) }
    after { travel_back }
```

I see that `around` is used elsewhere and it may not be as simple to split into a `before` and `after` part.

Reply:
Both good points, thanks. I've updated.

(The `after` block isn't actually needed - when `travel_to` is called in a block, `travel_back` is implicitly called when the block completes.)